pkg/pool-hooks/contracts/NftLiquidityPositionExample.sol

```diff
@@ -0,0 +1,257 @@
+                                 Hook Logic
+    ***************************************************************************/
+
+    /**
+     * @dev The pool does not support adding liquidity through donation.
+     * There is an existing similar error (IVaultErrors.DoesNotSupportDonation), but hooks should not throw
+     * "Vault" errors.
+     */
+    error PoolDoesNotSupportDonation();
+
+    function onRegister(
+        address,
+        address,
+        TokenConfig[] memory,
+        LiquidityManagement calldata liquidityManagement
+    ) public virtual returns (bool) {
+        // This hook requires donation support to work (see above).
+        if (liquidityManagement.enableDonation == false) {
+            revert PoolDoesNotSupportDonation();
+        }
+
+        return true;
+    }
+
+    function getHookFlags() public view virtual returns (HookFlags memory) {
+        HookFlags memory hookFlags;
+        // `enableHookAdjustedAmounts` must be true for all contracts that modify the `amountCalculated`
+        // in after hooks. Otherwise, the Vault will ignore any "hookAdjusted" amounts, and the transaction
+        // might not settle. (It should be false if the after hooks do something else.)
+        hookFlags.enableHookAdjustedAmounts = true;
+        hookFlags.shouldCallAfterRemoveLiquidity = true;
+        hookFlags.shouldCallBeforeAddLiquidity = true;
+        return hookFlags;
+    }
+
+    function onBeforeAddLiquidity(
+        address router,
+        address,
+        AddLiquidityKind,
+        uint256[] memory,
+        uint256,
+        uint256[] memory,
+        bytes memory
+    ) public virtual returns (bool) {
+        // We only allow addLiquidity via the Router/Hook itself (as it must custody BPT)
+        require(router == address(this), "Can't use external router");
+        return true;
+    }
+
+    function onAfterRemoveLiquidity(
+        address router,
+        address pool,
+        RemoveLiquidityKind,
+        uint256,
+        uint256[] memory,
+        uint256[] memory amountsOutRaw,
+        uint256[] memory,
+        bytes memory userData
+    ) public virtual returns (bool, uint256[] memory hookAdjustedAmountsOutRaw) {
+        // We only allow removeLiquidity via the Router/Hook itself so that fee is applied correctly
+        require(router == address(this), "Can't use external router");
+
+        uint256 tokenId = abi.decode(userData, (uint256));
+        hookAdjustedAmountsOutRaw = amountsOutRaw;
+        // Calculate the number of days that have passed since startTime
+        uint256 daysPassed = (block.timestamp - startTime[tokenId]) / 1 days;
+        // Initial fee of 10%
+        uint256 initialFee = 10e16;
+        if (daysPassed < 10) {
+            // decreasing fee by 1% per day
+            uint256 currentFee = initialFee - 1e16 * daysPassed;
+            hookAdjustedAmountsOutRaw = _takeFee(pool, amountsOutRaw, currentFee);
```

Comment:
Maybe a view function where I give an `id` and it returns what the remaining fee is.